oak_functions/loader/src/server.rs

```diff
@@ -40,35 +82,28 @@ impl WasmHandler {
         })
     }
 
-    pub(crate) fn handle_request(&self, req: &Request<Body>) -> anyhow::Result<Response<Body>> {
+    pub(crate) async fn handle_request(
+        &self,
+        req: Request<Body>,
+    ) -> anyhow::Result<Response<Body>> {
         info!("The request is: {:?}", req);
         match (req.method(), req.uri().path()) {
-            (&hyper::Method::POST, "/invoke") => self.handle_invoke(req),
+            (&hyper::Method::POST, "/invoke") => self.handle_invoke(req).await,
             (method, path) => http::response::Builder::new()
                 .status(StatusCode::BAD_REQUEST)
                 .body(format!("Invalid request: {} {}\n", method, path).into())
                 .context("Couldn't create response"),
         }
     }
 
-    fn handle_invoke(&self, _req: &Request<Body>) -> anyhow::Result<Response<Body>> {
-        // TODO(#1919): Make request body available to the Wasm module via ABI functions.
-        let instance = wasmi::ModuleInstance::new(&self.module, &wasmi::ImportsBuilder::default())
-            .context("failed to instantiate Wasm module")?
-            .assert_no_start();
-
-        let result = instance.invoke_export(MAIN_FUNCTION_NAME, &[], &mut wasmi::NopExternals);
-
-        info!(
-            "{:?}: Running Wasm module completed with result: {:?}",
-            std::thread::current().id(),
-            result
-        );
-
+    async fn handle_invoke(&self, req: Request<Body>) -> anyhow::Result<Response<Body>> {
+        let request_bytes = hyper::body::to_bytes(req.into_body()).await.unwrap();
+        let mut wasm_state = WasmState::new(&self.module, request_bytes.to_vec())?;
+        wasm_state.invoke();
         // TODO(#1919): Get the actual response from the Wasm module via ABI functions.
```

Comment:
nit: Move this comment to `WasmState` too.

Reply:
Done.